docs: refresh tools page and try details tabs (#7769)
langflow-ai/langflow: docs/docs/Components/components-tools.md

This component searches and retrieves papers from [arXiv.org](https://arXiv.org).

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| search_query | String | The search query for arXiv papers (for example, `quantum computing`) |
| search_type | String | The field to search in |

> | max_results | Integer | Maximum number of results to return |

Reviewer: Add periods to the descriptions.

**Outputs**

| Name | Type | Description |
|------|------|-------------|
| papers | List[Data] | List of retrieved arXiv papers |

</details>

## Astra DB Tool

The `Astra DB Tool` allows agents to connect to and query data from Astra DB collections.

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| Tool Name | String | The name used to reference the tool in the agent's prompt. |
| Tool Description | String | A brief description of the tool. This helps the model decide when to use it. |
| Collection Name | String | The name of the Astra DB collection to query. |
| Token | SecretString | The authentication token for accessing Astra DB. |
| API Endpoint | String | The Astra DB API endpoint. |
| Projection Fields | String | The attributes to return, separated by commas. Default: "*". |

> | Tool Parameters | Dict | Parameters the model needs to fill to execute the tool. For required parameters, use an exclamation mark (for example, `!customer_id`). |

Reviewer: Avoid parentheses.

| Static Filters | Dict | Attribute-value pairs used to filter query results. |
| Limit | String | The number of records to return. |

**Outputs**

| Name | Type | Description |
|------|------|-------------|
| Data | List[Data] | A list of [Data](/concepts-objects) objects containing the query results from the Astra DB CQL table. Each Data object contains the document fields specified by the projection fields. Limited by the `number_of_results` parameter. |
| Tool | StructuredTool | A LangChain StructuredTool object that can be used in agent workflows. Contains the tool name, description, argument schema based on partition and clustering keys, and the query function. |

</details>

## Bing Search API

This component allows you to call the Bing Search API.

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| bing_subscription_key | SecretString | Bing API subscription key |
| input_value | String | Search query input |
| bing_search_url | String | Custom Bing Search URL (optional) |

> | k | Integer | Number of search results to return |

Reviewer: Add periods and articles to the descriptions. For example, "A Bing API subscription key.".

**Outputs**

| Name | Type | Description |
|------|------|-------------|
| results | List[Data] | List of search results |

> | tool | Tool | Bing Search tool for use in LangChain |

Reviewer: Add periods and articles to the descriptions.

</details>

## Combinatorial Reasoner

This component runs Icosa's Combinatorial Reasoning (CR) pipeline on an input to create an optimized prompt with embedded reasons. For more information, see [Icosa computing](https://www.icosacomputing.com/).

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| prompt | String | Input to run CR on |
| openai_api_key | SecretString | OpenAI API key for authentication |
| username | String | Username for Icosa API authentication |
| password | SecretString | Password for Icosa API authentication |
| model_name | String | OpenAI LLM to use for reason generation |

**Outputs**

| Name | Type | Description |
|------|------|-------------|
| optimized_prompt | Message | A message object containing the optimized prompt |

> | reasons | List[String] | A list of the selected reasons that are embedded in the optimized prompt |

Reviewer: Add periods to the descriptions.

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| input_value | String | The search query to execute with DuckDuckGo. |
| max_results | Integer | The maximum number of search results to return. Default: `5`. |
| max_snippet_length | Integer | The maximum length of each result snippet. Default: `100`. |

**Outputs**

| Name | Type | Description |
|------|------|-------------|
| data | List[Data] | List of search results as Data objects containing snippets and full content. |
| text | String | Search results formatted as a single text string. |

</details>

## Exa Search

This component provides an [https://exa.ai/](Exa Search) toolkit for search and content retrieval.

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| metaphor_api_key | SecretString | API key for Exa Search (entered as a password) |
| use_autoprompt | Boolean | Whether to use autoprompt feature (default: true) |
| search_num_results | Integer | Number of results to return for search (default: 5) |

> | similar_num_results | Integer | Number of similar results to return (default: 5) |

Reviewer: Add periods and articles to the descriptions. Also, avoid parentheses.

**Outputs**

| Name | Type | Description |
|------|------|-------------|

> | tools | List[Tool] | List of search tools provided by the toolkit |

Reviewer: ```suggestion
| tools | List[Tool] | List of search tools that the toolkit provides. |
```

</details>

## Glean Search API

This component allows you to call the Glean Search API.

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| glean_api_url | String | URL of the Glean API |
| glean_access_token | SecretString | Access token for Glean API authentication |
| query | String | Search query input |
| page_size | Integer | Number of results per page (default: 10) |

> | request_options | Dict | Additional options for the API request (optional) |

Reviewer: Add periods and articles to the descriptions. Also, avoid parentheses.

**Outputs**

| Name | Type | Description |
|------|------|-------------|
| results | List[Data] | List of search results |

> | tool | Tool | Glean Search tool for use in LangChain |

Reviewer: Add periods and articles to the descriptions. Apply this throughout this PR. I won't comment on the rest that I find.

| Tool | StructuredTool | A LangChain `StructuredTool` object that can be used in agent workflows. Contains the tool name, description, argument schema based on tool parameters, and the query function. |

</details>

## Astra DB CQL Tool

The `Astra DB CQL Tool` allows agents to query data from CQL tables in Astra DB.

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| Tool Name | String | The name used to reference the tool in the agent's prompt. |
| Tool Description | String | A brief description of the tool to guide the model in using it. |
| Keyspace | String | The name of the keyspace. |
| Table Name | String | The name of the Astra DB CQL table to query. |
| Token | SecretString | The authentication token for Astra DB. |
| API Endpoint | String | The Astra DB API endpoint. |
| Projection Fields | String | The attributes to return, separated by commas. Default: "*". |
| Partition Keys | Dict | Required parameters that the model must fill to query the tool. |

> | Clustering Keys | Dict | Optional parameters the model can fill to refine the query. Required parameters should be marked with an exclamation mark (for example, `!customer_id`). |

Reviewer: ```suggestion
| Clustering Keys | Dict | Optional parameters the model can fill to refine the query. Required parameters should be marked with an exclamation mark, for example, `!customer_id`. |
```

6. Open the **Playground** and chat with your tool.
The agent chooses the correct tool based on your query.

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| command | String | MCP command (default: `uvx mcp-sse-shim@latest`) |

**Outputs**

| Name | Type | Description |
|------|------|-------------|
| tools | List[Tool] | List of tools exposed by the MCP server |

</details>

## Wikidata

This component performs a search using the Wikidata API.

<details>
<summary>Parameters</summary>

**Inputs**

| Name | Type | Description |
|------|------|-------------|
| query | String | The text query for similarity search on Wikidata. |

**Outputs**

| Name | Type | Description |
|------|------|-------------|
| data | List[Data] | The search results from Wikidata API as a list of Data objects. |
| text | Message | The search results formatted as a text message. |

</details>

## Legacy components

> Legacy components are available to use but no longer supported.

Reviewer: ```suggestion
Legacy components are available for use but are no longer supported.
```
